File: leetcode/subArraySumEqualsK.cpp
class Solution {
public:
    
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> myMap;
        int ans = 0;
        int sum = 0;
        int i = 0;
        int n = nums.size();
        
        myMap[sum] = 1;
        
        while(i < n) {
            sum += nums[i];
            int find = sum - k;
            
            if(myMap.find(find) != myMap.end()) ans += myMap[find];
            
            myMap[sum]++;
            i++;
            
        }
        
        return ans;
    }
};

File: leetcode/subArraySumEqualsK.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
 var subarraySum = function(nums, k) {
  let myMap = new Map()
  
  let ans = 0
  let sum = 0
  
  myMap.set(0, 1)
  
  for(let i = 0; i < nums.length; i++) {
      sum += nums[i]
      let find = sum - k
      
      if(myMap.get(find) != undefined) ans += myMap.get(find)
      
      if(myMap.get(sum) == undefined) myMap.set(sum, 1)
      else myMap.set(sum, myMap.get(sum) + 1)
  }
  
  return ans
};